maxfw/utils/image_utils.py

```diff
@@ -0,0 +1,190 @@
+    """
+
+    def __init__(self, transforms=[]):
+        assert isinstance(transforms, Iterable)
+        self.transforms = transforms
+
+    def apply_transforms(self, img):
+        for t in self.transforms:
+            img = t(img)
+        return img
+
+
+class ToPILImage(object):
+    """Convert a byte stream or an ndarray to PIL Image.
+
+    Converts a byte stream or a numpy ndarray of shape
+    H x W x C to a PIL Image while preserving the value range.
+
+    Args:
+        mode (`PIL.Image mode`_): color space and pixel depth of input data (optional).
+            If ``mode`` is ``None`` (default) there are some assumptions made about the input data:
+             - If the input has 4 channels, the ``mode`` is assumed to be ``RGBA``.
+             - If the input has 3 channels, the ``mode`` is assumed to be ``RGB``.
+             - If the input has 2 channels, the ``mode`` is assumed to be ``LA``.
+             - If the input has 1 channel, the ``mode`` is determined by the data type (i.e ``int``, ``float``,
+              ``short``).
+
+    .. _PIL.Image mode: https://pillow.readthedocs.io/en/latest/handbook/concepts.html#concept-modes
+    """
+    def __init__(self, target_mode, mode=None):
+        self.mode = mode
+        self.target_mode = target_mode
+
+    def __call__(self, pic):
+        """
+        Args:
+            pic (bytestream or numpy.ndarray): Image to be converted to PIL Image.
+
+        Returns:
+            PIL Image: Image converted to PIL Image.
+
+        """
+        return F.to_pil_image(pic, self.target_mode, self.mode)
+
+
+class Normalize(object):
+    """
+    Normalize the image to a range between [0, 1].
+    """
+
+    def __call__(self, img):
+        """
+        Args:
+        img (PIL image or numpy.ndarray): Image to be normalized.
+
+        Returns:
+        numpy.ndarray: Normalized image.
+        """
+        return F.normalize(img)
+
+
+class Standardize(object):
+    """
+    Standardize the image (mean-centering and STD of 1).
+    """
+
+    def __call__(self, img):
+        """
+        Args:
+        img (PIL image or numpy.ndarray): Image to be standardized.
+
+        Returns:
+        numpy.ndarray: Standardized image.
+        """
+        return F.standardize(img)
+
+
+class Resize(object):
+    """Resize the input PIL Image to the given size.
+
+    Args:
+        size (sequence or int): Desired output size. If size is a sequence like
+            (h, w), output size will be matched to this. If size is an int,
+            smaller edge of the image will be matched to this number.
+            i.e, if height > width, then image will be rescaled to
+            (size * height / width, size)
+        interpolation (int, optional): Desired interpolation. Default is
+            ``PIL.Image.BILINEAR``
+    """
+
+    def __init__(self, size, interpolation=Image.BILINEAR):
+        assert isinstance(size, int) or (isinstance(size, Iterable) and len(size) == 2)
```

Comment:
For the second half, do you mean `isinstance(size, Sequence)`?

Reply:
I guess that would be more specific, thanks. Will have the change in the next update.

Reply:
For my understanding, if the size would be a string, it would qualify as an `Iterable` but not as a `Sequence` right? @xuhdev

Reply:
```
In [2]: from collections.abc import Iterable, Sequence

In [3]: isinstance('abc', Iterable)
Out[3]: True

In [4]: isinstance('abc', Sequence)
Out[4]: True

```

My understanding is that all `Sequence`s are `Iterable`s. A `Sequence` always has everything in place and can be visited at any time, but an `Iterable` does not necessarily do. The behavior of applying `len` to or go through an `Iterable` that is not a `Sequence` is not guaranteed. For example,

```
In [1]: from collections.abc import Iterable, Sequence

In [2]: my_iter = (i for i in range(100))

In [3]: isinstance(my_iter, Iterable)
Out[3]: True

In [4]: isinstance(my_iter, Sequence)
Out[4]: False

In [5]: len(my_iter)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-5-f34562eb0975> in <module>
----> 1 len(my_iter)

TypeError: object of type 'generator' has no len()
```

Reply:
Interesting, thanks @xuhdev for clarifying!

Reply:
@kmh4321 If you're interested